libs/damap/src/lib/interceptors/auth.interceptor.ts

```diff
@@ -0,0 +1,41 @@
+import { Injectable } from '@angular/core';
+import {
+  HttpInterceptor,
+  HttpRequest,
+  HttpHandler,
+  HttpEvent,
+} from '@angular/common/http';
+import { Observable } from 'rxjs';
+import { AuthService } from '../auth/auth.service';
+import { FeedbackService } from '../services/feedback.service';
+import { OAuthService } from 'angular-oauth2-oidc';
+
+@Injectable()
+export class AuthInterceptor implements HttpInterceptor {
+  constructor(
+    private authService: AuthService,
+    private oAuthService: OAuthService,
+    private feedbackService: FeedbackService,
+  ) {}
+
+  intercept(
+    req: HttpRequest<any>,
+    next: HttpHandler,
+  ): Observable<HttpEvent<any>> {
+    const accessToken = this.oAuthService.getAccessToken();
+    if (accessToken == null) {
+      return next.handle(req);
+    }
+
+    const date = new Date(this.oAuthService.getAccessTokenExpiration());
+    const timeLeft = date.getTime() - new Date().getTime();
```

Comment:
`AuthService` has a method called `isAuthenticated` . Would this also perform an expiration check?